### PR TITLE
Edgecloud 4350 2.4.1

### PR DESCRIPTION
**Pull request by lgarner**
- Fix too vague connection error messages (RegisterClient)
- Extra guard mode to prevent Permission dialogs when not needed by application.
- DLL update needed as well (TBD).

### CODE REVIEW

File: Runtime/Scripts/MobiledgeXIntegrationHelper.cs [hunk 1]
```diff
@@ -344,6 +344,12 @@ public async Task<bool> IsRoaming()
             string wifiIpV4 = null;
             string wifiIpV6 = null;
 
+            if (!matchingEngine.EnableEnhancedLocationServices)
```
Comment (ah1053) [inline Runtime/Scripts/MobiledgeXIntegrationHelper.cs:347]: EnableEnhancedLocationServices will be sat in the Dll then?

Comment (lgarner) [reply Runtime/Scripts/MobiledgeXIntegrationHelper.cs:347]: Just included the DLL with the fix to avoid the local network permissions.

**Review by ah1053 — APPROVED** [review]
tested on iOS 14.3 and local network  permission didn't appear.
Thank you Garner 🎩

**Comment by ah1053** [comment]
Can you change the base to staging 2.5 
If you don't want to change replace Debug with Logger, I can't do it with the ChangeLog PR

**Pull request merged**